### PR TITLE
Remove legacy shutdown

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11352/

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11356/

File: arangod/Aql/EngineInfoContainerDBServerServerBased.cpp [hunk 1]
```diff
@@ -379,8 +379,7 @@ Result EngineInfoContainerDBServerServerBased::buildEngines(
     if (!result.ok()) {
       return result;
     }
-    auto insert = serverToQueryId.try_emplace(serverDest, globalId);
-    TRI_ASSERT(insert.second);
+    serverToQueryId.emplace_back(serverDest, globalId);
```
Comment (jsteemann) [inline arangod/Aql/EngineInfoContainerDBServerServerBased.cpp:382]: micro optimization: we can make `serverDest` non-const and move it into `serverToQueryId`.
```suggestion
    serverToQueryId.emplace_back(serverDest, globalId);
```

File: arangod/Aql/ExecutionEngine.cpp [hunk 2]
```diff
@@ -475,100 +467,57 @@ struct DistributedQueryInstanciator final
...
-        }
-      } else {
-        engine->setShutdown(ExecutionEngine::ShutdownState::Legacy);
-      }
-    }
+//    for (auto& engine : snippets) {
+//      if (engine->engineId() != 0) { // only the root snippet executes a shutdown on the Coordinator
+//        engine->setShutdown(ExecutionEngine::ShutdownState::Done);
+//      }
+//    }
```
Comment (jsteemann) [inline arangod/Aql/ExecutionEngine.cpp:510]: this can possibly be removed now.

File: arangod/Aql/ExecutionEngine.cpp [hunk 3]
```diff
@@ -475,100 +467,57 @@ struct DistributedQueryInstanciator final
...
     
-    TRI_ASSERT(snippets[0].first == 0);
-    snippets[0].second->snippetMapping(std::move(snippetIds), std::move(serverToQueryId));
-    snippets[0].second->globalStats().setAliases(std::move(nodeAliases));
-    TRI_ASSERT(snippets[0].second->shutdownState() != ExecutionEngine::ShutdownState::Done);
+    TRI_ASSERT(snippets[0]->engineId() == 0);
+//    snippets[0]->snippetMapping(std::move(snippetIds), std::move(serverToQueryId));
+    _query.executionStats().setAliases(std::move(nodeAliases));
+//    TRI_ASSERT(snippets[0]->shutdownState() != ExecutionEngine::ShutdownState::Done);
```
Comment (jsteemann) [inline arangod/Aql/ExecutionEngine.cpp:515]:
```suggestion
    _query.executionStats().setAliases(std::move(nodeAliases));
```

File: arangod/Aql/ExecutionEngine.cpp [hunk 4]
```diff
@@ -669,91 +618,74 @@ std::pair<ExecutionState, size_t> ExecutionEngine::skipSome(size_t atMost) {
...
-  }
-
-  return res;
-  
-} catch (...) {
-  // nothing we can do here...
-  return Result(TRI_ERROR_INTERNAL, "unable to shutdown query");
-}
-
+//Result ExecutionEngine::shutdownSync(int errorCode) noexcept try {
```
Comment (jsteemann) [inline arangod/Aql/ExecutionEngine.cpp:621]: All this can hopefully be removed now?

File: arangod/Aql/ExecutionEngine.cpp [hunk 5]
```diff
@@ -669,91 +618,74 @@ std::pair<ExecutionState, size_t> ExecutionEngine::skipSome(size_t atMost) {
...
-    LOG_TOPIC("47b85", INFO, Logger::QUERIES) << "executing legacy query shutdown";
-    
-    _shutdownState.store(ShutdownState::Done, std::memory_order_relaxed);
-    
-    return {state, res};
-  }
-  
-  return {ExecutionState::DONE, _shutdownResult};
-}
+//std::pair<ExecutionState, Result> ExecutionEngine::shutdown(int errorCode) {
```
Comment (jsteemann) [inline arangod/Aql/ExecutionEngine.cpp:658]: same here, I hope we can remove the code now.

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11373/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11374/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11375/

File: arangod/Aql/QueryCursor.cpp [hunk 6]
```diff
@@ -198,11 +200,12 @@ QueryStreamCursor::~QueryStreamCursor() {
...
+  // remove the continue handler we may have registered in the query
+  _query->sharedState()->resetWakeupHandler();
 
-  _queryResults.clear();
-
-  // now remove the continue handler we may have registered in the query
-  _query->sharedState()->invalidate();
+  while (!_queryResults.empty()) {
+    _queryResults.pop_front();
+  }
```
Comment (jsteemann) [inline arangod/Aql/QueryCursor.cpp:208]: Wouldn't `_queryResults.clear()` be simpler here?

**Comment by jsteemann** [comment]
Test failures, including an assertion failure, in http://172.16.10.101:8080/job/arangodb-matrix-pr-linux/13046/

**Comment by graetzer** [comment]
blue http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11389/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11404/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**